tests/sdk/test_workflow.py

```diff
@@ -399,3 +399,24 @@ def my_workflow():
         pass
 
     assert my_workflow._default_dependency_imports == expected_imports
+
+
+def test_with_resources_copy_imports():
```

Comment:
Suggestion: I think the following is slightly clearer, but that might just be me :)

```suggestion
def test_with_resources_copies_imports():
```